### DISCUSSION TITLE
Offer a Gradle Plugin

### DISCUSSION THREAD

**Discussion by desiderantes** [bailuk/java-gtk#32] [General]
I'm very interested in this library and I'd like to know if there's any interest in offering a Gradle plugin to ease the code generation aspect. Of course I'm also interested in helping with code if needed.

**Comment by bailuk** [comment]
I started with JNI before I switched to JNA. The Makefile "Wrapper" gave me the flexibility to automatically build the C-Layer. I still left it for convenience, but I'm not using it often anymore. So its something that I will eventually remove.

A Gradle plugin that runs the code generator when needed (For example when the generator itself has changed) would be very useful and welcome.

Its important for me that additions are low on maintenance and that they use as few as possible (preferably none) external dependencies. And they need to work on all environments (Including Windows, MacOS and Nix)

**Comment by bailuk** [comment]
I've moved this to discussions, as its not really an issue
